test/e2e.sh

```diff
@@ -0,0 +1,80 @@
+#! /bin/bash
+
+set -eux -o pipefail 
+
+# Get running RHCOS version. Assumes all nodes are running the same version.
+# TODO check RT kernel version
+get_rhcos_version() {
+    worker_node=$(kubectl get nodes --selector='node-role.kubernetes.io/worker' -ojsonpath='{.items[].metadata.name}')
+    rhcos_version=$(oc debug node/${worker_node} \
+	    --quiet \
+	    -- cat /host/etc/os-release \
+	    | grep "OSTREE_VERSION" \
+	    | cut -d= -f2 \
+	    | tr -d \'\")
+
+    echo ${rhcos_version}
+}
+
+# Check if driver-toolkit imagestream is available
+test_imagestream() {
+    oc get imagestream/driver-toolkit -n openshift
+}
+
+# Check for latest and RHCOS tags
+test_imagestream_tags() {
+    rhcos_version=$(get_rhcos_version)
+    echo "INFO: Node RHCOS version: ${rhcos_version}"
+
+    oc get imagestream/driver-toolkit -n openshift -o json | jq ".spec.tags[].name"
+    
+    image_latest=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"latest\") | .from.name")
+    echo "INFO: Image from latest tag: ${image_latest}"
+        
+    image_rhcos_tag=$(oc get imagestream/driver-toolkit -n openshift -o json \
+    	| jq ".spec.tags[] | select(.name == \"${rhcos_version}\") | .from.name")
+    echo "INFO: Image from RHCOS tag: ${image_rhcos_tag}"
+
+    [[ $image_latest != "" ]]
+
+    [[ $image_rhcos_tag != "" ]]
+    
+    [[ $image_rhcos_tag == $image_latest ]]
+}
+
+# Check that DTK contains the right kernel version
+test_kernel_version() {
+    node_kernel=$(oc get nodes -ojsonpath='{.items[].status.nodeInfo.kernelVersion}')
+    echo "INFO: Node kernel: ${node_kernel}"
+
+    dtk_release=$(oc debug --image-stream="openshift/driver-toolkit:latest" \
+	    --quiet \
+            -- cat /etc/driver-toolkit-release.json)
+    dtk_kernel=$(echo "${dtk_release}" | jq -r .KERNEL_VERSION)
+    echo "INFO: driver-toolkit kernel: ${dtk_kernel}"
+    
+    [[ ${dtk_kernel} ==  ${node_kernel} ]]
```

Comment:
hum, think it could be clearer to add a test here,
```
if [[ ${dtk_kernel} !=  ${node_kernel} ]]; then
  echo "ERROR: dtk and node kernel version don't match (${dtk_kernel} !=  ${node_kernel})"
  exit 1
fi
```
this way you cover "expected" errors (= OCP errors)

Reply:
Thanks, I used an if statement like this for the primary test cases.